src/core/view_manager.cpp

```diff
@@ -923,10 +926,10 @@ void ViewManager::propose_changes(DerechoSST& gmsSST) {
         gmssst::set(gmsSST.changes[my_rank][last_change_index].end_of_view, false);
     } else {
         //Push all the proposed changes, including joiner information if any joins were proposed
-        gmsSST.put(gmsSST.changes);
+        gmsSST.put_with_completion(gmsSST.changes);
```

Comment:
I don't see why we need to change the put call to put_with_completion. Are you just trying to safe? These protocols work by the nodes acknowledging these changes in multiple steps. So waiting for completion is redundant.